PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs

```diff
@@ -1099,8 +1099,38 @@ public void ExecutePositionsAction(Action<ShapeRange> positionsAction, bool isPr
             {
                 if (simulatedShapes != null)
                 {
-                    simulatedShapes.Delete();
-                    GC.Collect();
+                    try
+                    {
+                        simulatedShapes.Delete();
+                        GC.Collect();
+                    }
+                    // Catch corrupted shapes
+                    catch (System.Runtime.InteropServices.COMException)
+                    {
+                        // Remove all simulated shapes manually
+                        for (int i = 0; i < simulatedShapes.Count; i++)
+                        {
+                            // This method to remove duplicated shapes might fail for non-corrupted shapes when mixing good/bad shapes
+                            if (this.GetCurrentSlide().Shapes[1].Name.Contains("_Copy"))
```

Comment:
Will this accidentally delete shapes that should not be affected by this operation?

Reply:
Is the logic correct? Thought it should check shapes[i] instead of [1] everytime

Reply:
duplicated shapes are somehow appended to the starts of the Shapes collection, hence they are referred to by shapes[1]. If you index it with i instead, it may delete the nonduplicated shapes at the back. Think of this as popping from the stack.

Furthermore, it is not code practice to delete shapes by their indexes while in a for loop, because it will mess up the delete operations

Reply:
It has been tested and did not delete other unintended shapes.

I have tried to make it work as best as i could. The operation works well for when Distribute is performed for a set of "good" shapes and a set of "bad" shapes, but may be flaky for when there is a mixture of good/bad shapes. I have tried to solve this but faced API limitations. 

Hence the error message (Like for Radial and Grid) to get the user to press undo to ensure that state of the slides is reverted back to original.

Reply:
Hm then I think it should break out of loop as soon as it does not contain "Copy"? Otherwise it seems to be checking the same shape multiple times

Reply:
I have added the breaks for the for loops. Thanx for that spot!